src/astro/databases/snowflake.py

```diff
@@ -57,6 +81,321 @@ def get_table_qualified_name(table: Table) -> str:  # skipcq: PYL-R0201
         qualified_name = ".".join(name for name in qualified_name_lists if name)
         return qualified_name
 
+    def check_optimised_path(self, source_file: File, target_table: Table) -> bool:
+        """
+        Check if there is an optimised path for source to destination.
+
+        :param source_file: File from which we need to transfer data
+        :param target_table: Table that needs to be populated with file data
+        """
+        file_type = self.optimized_path_supported_file_types.get(source_file.type.name)
```

Comment:
@sunank200 - Maybe not right now, can these checks can partially or completely be dependent on different sources? Like for S3. Like we may require 'storage_integration' for one and not for another or some different way to authenticate.